driver-core/src/test/unit/com/mongodb/internal/connection/TestConnectionPoolListener.java

```diff
@@ -29,23 +29,40 @@
 import com.mongodb.event.ConnectionReadyEvent;
 
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.locks.Condition;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
 @SuppressWarnings("deprecation")
 public class TestConnectionPoolListener implements ConnectionPoolListener {
 
+    private final Set<String> eventTypes;
+
     private final List<Object> events = new ArrayList<Object>();
     private final Lock lock = new ReentrantLock();
     private final Condition condition = lock.newCondition();
     private volatile Class<?> waitingForEventClass;
     private volatile int waitingForEventCount;
+    private final AtomicInteger numConnectionsCheckedOut = new AtomicInteger();
+
 
     public TestConnectionPoolListener() {
+        this(Arrays.asList("poolCreatedEvent", "poolReadyEvent", "poolClearedEvent", "poolClosedEvent", "connectionCreatedEvent",
+                "connectionReadyEvent", "connectionClosedEvent", "connectionCheckOutStartedEvent", "connectionCheckOutFailedEvent",
+                "connectionCheckedOutEvent", "connectionCheckedInEvent",
+                // These are deprecated, but still used by some tests
+                "poolOpenedEvent", "connectionAddedEvent", "connectionRemovedEvent"));
+    }
+
+    public TestConnectionPoolListener(final List<String> eventTypes) {
```

Comment:
What is the reason to accept `List<String>` instead of `Set<String>`? It appears to me that accepting the latter is more convenient here.